### PR TITLE
Use GoldToken for web faucet so the transfers show up in the feed

### LINKED ISSUE #152: Gold faucet transactions don’t show up in the feed
### Expected Behavior

In feed

### Current Behavior

Gold faucet transactions don’t show up in the feed

### PR DESCRIPTION
**Pull request by yerdua**
### Description

Gold transfers that happened via the web faucet were not showing up in the feed in the wallet app. The reason for this is that the faucet used a native transfer for gold, and the wallet app is only set up to display token transfers done with the token contracts. 

This commit fixes the problem by modifying the faucet cloud function to use the GoldToken contract instead of transferring gold natively. This solution was chosen because despite using more gas, it avoids extra complexity or doubling the calls to Blockscout in the wallet app or blockchain-api.

### Tested

This hasn't yet been deployed to firebase, but was tested locally with this command and confirming that the transfer was displayed in the feed for my test account.

`yarn transfer-funds --nodeUrl http://35.185.236.10:8545 --stableTokenAddress 0x299E74bdCD90d4E10f7957EF074ceE32d7e9089a --goldTokenAddress 0x4813BFD311E132ade22c70dFf7e5DB045d26D070 --gold 10000000000000000000 --dollar 10000000000000000000 932f5d5ec576ad523e6940f0bc21aeeb37f90f864ba714462fcdd1d13c6d5399 0xa7ee7f12f9f636dabe39e1f8428bd16493518742`

### Other changes

The types in `BlockscoutTransaction` have been updated to reflect the real types that are returned from the api (all the "numbers" are actually strings). This necessitated parsing those string values with BigNumber.

I also included a comment about why native transactions aren't returned, explaining the tradeoff, and adding a reference to #226

### Related issues

- Fixes #152

### CODE REVIEW

**Review by jmrossy — APPROVED** [review]
Looks good to me! Would be good to get @mcortesi approval as well before merging :)

**Review by mcortesi — APPROVED** [review]
lgtm!

**Pull request merged**
